### PR TITLE
feat: Support for directory recursion for ArgoCD app added

### LINKED ISSUE #73: Add recursion support for ArgoCD Applications
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

#### What is the outcome that you are trying to reach?

Sometimes there are a lot of various resources in a repository (e.g. apps of apps for different environments/domains/lines of business) and it's more convenient to put them to corresponding subdirectories.

#### Describe the solution you would like

ArgoCD has built-in support for recursion: https://argo-cd.readthedocs.io/en/stable/user-guide/directory/#enabling-recursive-resource-detection
Let's allow users to enable it for certain apps

#### Describe alternatives you have considered

The only alternative I can think of would be creating multiple apps in ArgoCD, which might be not convenient

#### Additional context

The link above describes this feature, I believe :)

### PR DESCRIPTION
**Pull request by melnikovn**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints-addons/blob/main/.github/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

This PR adds support for ArgoCD app recursion: https://argo-cd.readthedocs.io/en/stable/user-guide/directory/#enabling-recursive-resource-detection

### Motivation

- Resolves #73

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I ran `pre-commit run -a` with this PR

### For Moderators

- [ ] E2E Test successfully complete before merge?

### Additional Notes

The following app was created as a result:

```
project: default
source:
  repoURL: '.......'
  path: cluster-staging/argocd-apps/envs
  targetRevision: main
  directory:
    recurse: true
    jsonnet: {}
destination:
  server: 'https://kubernetes.default.svc'
  namespace: argocd
syncPolicy:
  automated:
    prune: true
    selfHeal: true
  syncOptions:
    - Validate=false
    - CreateNamespace=true
    - PrunePropagationPolicy=foreground
    - PruneLast=true
    - RespectIgnoreDifferences=true
  retry:
    limit: 10
    backoff:
      duration: 10s
      factor: 2
      maxDuration: 3m
```

### CODE REVIEW

File: modules/argocd/main.tf [hunk 1]
```diff
@@ -96,6 +96,12 @@ resource "helm_release" "argocd_application" {
     })
   ]
 
+  set {
+    name  = "source.useRecurse"
+    value = each.value.use_recurse
+    type  = "boolean"
+  }
+
```
Comment (askulkarni2) [inline modules/argocd/main.tf:104]: As per https://argo-cd.readthedocs.io/en/stable/user-guide/directory/#enabling-recursive-resource-detection, directory-type applications only work for plain manifest files. So this doesn't seem to be valid.

**Review by askulkarni2 — CHANGES_REQUESTED** [review]

File: modules/argocd/main.tf (hunk 1 above)
Comment (melnikovn) [reply modules/argocd/main.tf:104]: correct... let me double check.
my original pr did not have this part, but afair pre-commit was complaining.
i'll work on it a bit more

Comment (melnikovn) [reply modules/argocd/main.tf]: updated

**Review by askulkarni2 — APPROVED** [review]
LGTM! Thanks for the PR!

**Pull request merged**